aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -258,25 +249,236 @@ The following component:
+
+```csharp
+[Parameter]
+[SupplyParameterFromQuery(Name = "{QUERY PARAMETER NAME}")]
+public string {COMPONENT PARAMETER NAME} { get; set; }
+```
+
+In the following example with a URL of `/search?filter=scifi%20stars&page=3&star=LeVar%20Burton&star=Gary%20Oldman`:
+
+* The `Filter` property resolves to `scifi stars`.
+* The `Page` property resolves to `3`.
+* The `Stars` array is filled from query parameters named `star` (`Name = "star"`) and resolves to `LeVar Burton` and `Gary Oldman`.
+
+`Pages/Search.razor`:
+
+```razor
+@page "/search"
+
+<h1>Search Example</h1>
+
+<p>Filter: @Filter</p>
+
+<p>Page: @Page</p>
+
+<p>Assignees:</p>
+
+<ul>
+    @foreach (var name in Stars)
+    {
+        <li>@name</li>
+    }
+</ul>
+
+@code {
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public string Filter { get; set; }
+
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public int? Page { get; set; }
+
+    [Parameter]
+    [SupplyParameterFromQuery(Name = "star")]
+    public string[] Stars { get; set; }
+}
+```
+
+Use `NavigationManager.GetUriWithQueryParameter` to add, change, or remove one or more query parameters on the current URL:
 
 ```razor
 @inject NavigationManager NavigationManager
 
 ...
 
-var query = new Uri(NavigationManager.Uri).Query;
+NavigationManager.GetUriWithQueryParameter("{NAME}", {VALUE})
 ```
 
-To parse a query string's parameters, one approach is to use [`URLSearchParams`](https://developer.mozilla.org/docs/Web/API/URLSearchParams) with [JavaScript (JS) interop](xref:blazor/js-interop/call-javascript-from-dotnet):
+For the preceding example:
 
-```javascript
-export createQueryString = (string queryString) => new URLSearchParams(queryString);
+* The `{NAME}` placeholder specifies the query parameter name. The `{VALUE}` placeholder specifies the value as a supported type. Supported types are listed later in this section.
+* A string is returned equal to the current URL with a single parameter:
+  * Added if the query parameter name doesn't exist in the current URL.
+  * Updated to the value provided if the query parameter exists in the current URL.
+  * Removed if the type of the provided value is nullable and the value is `null`.
+* The correct culture-invariant formatting is applied for the given type (<xref:System.Globalization.CultureInfo.InvariantCulture?displayProperty=nameWithType>).
+* The query parameter name and value are URL-encoded.
+* All of the values with the matching query parameter name are replaced if there are multiple instances of the type.
+* If the value is an `IEnumerable<T>` for one of the supported types, then multiple name/value pairs are emitted. Supported types are listed later in this section.
```

Comment:
Actually, we removed the `IEnumerable<T>` APIs for `GetUriWithQueryParameter`. But, `GetUriWithQueryParameters` (notice the "s") does still support `IEnumerable<T>` values.